daml-assistant/daml-sdk/validate.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+# Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# --- begin runfiles.bash initialization v2 ---
+# Copy-pasted from the Bazel Bash runfiles library v2.
+set -uo pipefail; f=bazel_tools/tools/bash/runfiles/runfiles.bash
+source "${RUNFILES_DIR:-/dev/null}/$f" 2>/dev/null || \
+  source "$(grep -sm1 "^$f " "${RUNFILES_MANIFEST_FILE:-/dev/null}" | cut -f2- -d' ')" 2>/dev/null || \
+  source "$0.runfiles/$f" 2>/dev/null || \
+  source "$(grep -sm1 "^$f " "$0.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+  source "$(grep -sm1 "^$f " "$0.exe.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+  { echo>&2 "ERROR: cannot find $f"; exit 1; }; f=; set -e
+# --- end runfiles.bash initialization v2 ---
+
+JAVA=$(rlocation "$TEST_WORKSPACE/$1")
+SDK_CE=$(rlocation "$TEST_WORKSPACE/$2")
+SDK_EE=$(rlocation "$TEST_WORKSPACE/$3")
+
+for cmd in sandbox sandbox-classic; do
+    ! $JAVA -jar $SDK_CE $cmd --help | grep -q profile-dir
```

Comment:
Have you seen this fail? Without `-e` my understanding is that Bash will completely ignore all but the last iteration of the second loop.

Reply:
so if I replace this with `exit 1` it actually does fail so I think `!` doesn’t do what I thought it does.

Reply:
ahrg apparently `!` negates `set -e`. I hate bash so much.

Reply:
That copy-pasted segment is hard to parse. Maybe add an explicit `set -euo pipefail` after it? AFAIK `set` is idempotent. I'm also not quite sure about the priorities of `!` and `|`; it may be worth adding parentheses and/or moving the exclamation mark to the grep command.

Reply:
You could expend a few more characters and just be explicit:
```
for cmd in sandbox sandbox-classic; do
    if ! $JAVA -jar $SDK_CE $cmd --help | grep -q profile-dir; then
        exit 1
    fi
done
```